### PR TITLE
Reimplement history using Elm debugger's data structure

### PR DESCRIPTION
**Pull request by gigobyte**
this pull-request represents work on #8

### CODE REVIEW

File: src/Debug/Browser/History.elm [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
+    , currentSnapshot : Snapshot model msg
     }
 
 
-type History model msg
-    = History (Model model msg)
+type alias Snapshot model msg =
+    { model : model
+    , messages : Array msg
+    , messageCount : Int
```
Comment (opvasger) [inline src/Debug/Browser/History.elm]: I think you want to move the `messageCount` from `Snapshot` to `History` so the data-structure only stores the message-length of `currentSnapshot`. I'm assuming every `Snapshot`in `oldSnapshots` will have the same length as `snapshotSize`?

Comment (gigobyte) [reply src/Debug/Browser/History.elm]: Sure, but I was relucant to add another another type just to have one less `Int` property.

**Comment by gigobyte** [comment]
>Is it more performant to initialize the array in a new Snapshot to snapshotSize instead of using Array.empty?
Array in Elm is not a JS array (it's a HAMT, I believe), so there's no way/point to initialise it with a fixed size. Arrays are slower than Lists for updating, but that's why we have a size cap.

File: src/Debug/Browser/History.elm (hunk 1 above)
Comment (opvasger) [reply src/Debug/Browser/History.elm]: well, it's one less int property for **every** `oldSnapshot`, no?

**Pull request merged**